pkg/ebpf/c/tracee.bpf.c

```diff
@@ -4969,25 +4969,32 @@ statfunc enum event_id_e net_packet_to_net_event(net_packet_t packet_type)
     return MAX_EVENT_ID;
 }
 
-// The address of &neteventctx->eventctx will be aligned as eventctx is the
-// first member of that packed struct. This is a false positive as we do need
-// the neteventctx struct to be all packed.
+// The address of &neteventctx->eventctx will be aligned as eventctx is the first member of that
+// packed struct. This is a false positive as we do need the neteventctx struct to be all packed.
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Waddress-of-packed-member"
 
 statfunc int should_submit_net_event(net_event_context_t *neteventctx,
-                                                   net_packet_t packet_type)
+                                     net_packet_t packet_type)
 {
-    // configure network events that should be sent to userland
-    if (neteventctx->md.submit & packet_type)
-        return 1;
+    // TODO after v0.15.0: After some testing, the caching is never used, as the net context is
+    // always a new one (created by the cgroup/skb program caller, AND there is a single map check
+    // for each protocol, each protocol check for submission. Go back to changes made by commit
+    // #4e9bb610049 ("network: ebpf: lazy submit checks for net events"), but still using enum and
+    // better code (will improve the callers syntax as well).
 
-    if (should_submit_by_ctx(net_packet_to_net_event(packet_type), &(neteventctx->eventctx))) {
-        neteventctx->md.submit |= packet_type;
-        // done, result cached for later.
-        return 1;
-    }
-    return 0;
+    enum event_id_e evt_id = net_packet_to_net_event(packet_type);
+
+    // Check if event has to be sumitted: if any of the policies matched, submit. All network events
+    // are base events (to be derived in userland). This means that if any of the policies need a
+    // network event, kernel can submit and let userland deal with it.
+
+    event_config_t *evt_config = bpf_map_lookup_elem(&events_map, &evt_id);
+    if (evt_config == NULL)
+        return 0;
+
+    // Any policy matched is enough to submit the net event
+    return evt_config->submit_for_policies;
```

Comment:
This is the configuration of the event, seems that no matching is taken into account now

Reply:
It should whenever I start applying kernel filters to network context (Which isn't the case YET). But it will.

Reply:
I'll probably create the same logic we do have for should_filter() but targeting network filters (to be done in kernel). Then I should consider the masking because the in-kernel filtering might have cleaned the ctx->matched_policies (which is not the case yet).

Reply:
But is it correct?
The scope was checked for this task and saved in matched scopes.
Here you don't take the scope into account only the events. What if the scope only wants events for e.g. the ping process and nothing else? Does it work as is now?

Reply:
Let me also give an example. 
Say that current process is 'ls' and you have a global policy that traces some network event, e.g. net_packet_ipv4, so match_policies check pass.
Also say you have a policy for ping that wants icmp events.
Now the icmp event is not part of the global policy, yet you will provide matched policy for the ping process. Am I wrong?

Reply:
Yes you are correct indeed. We should do what you're saying. Thanks for the attention call for this, you're correct.

Reply:
@yanivagman spoke to Geyslan, he will add a patch AND'ing the matched policies from "network context" (which is 1:1 to the task context using the socket). I'll test with the network profiles I have and we can merge that.

Reply:
We're not updating the event matched policies for the network events as we don't have (yet) ni-kernel filters for those. Once I add, then I'll change this function to reflect the in kernel filters in the mask as well.

Reply:
https://github.com/aquasecurity/tracee/pull/3198